src/scheduler/Scheduler.cpp

```diff
@@ -500,72 +478,69 @@ int Scheduler::scheduleFunctionsOnHost(const std::string& host,
 void Scheduler::callFunction(faabric::Message& msg, bool forceLocal)
 {
     // TODO - avoid this copy
```

Comment:
AFAICT we are not copying anymore? Could we remove the `TODO`?

Reply:
I'm pretty sure `*req->add_messages() = msg;` is doing a copy. If this function weren't so widely used I'd get rid of it altogether; all scheduling of functions should be done through `callFunctions`. Fortunately it's mostly just used in tests, and I'll port the important bits of Faasm (like the function chaining calls) to use `callFunctions`.